File: Domotización - Interfaces/Domotización - Interfaces/PuertaAutomatica.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domotización___Interfaces
{
    internal class PuertaAutomatica : Puerta, IElementoDomotizado
    {
        public PuertaAutomatica(int numPuerta) : base(numPuerta)
        {
        }

        public override void bloquear()
        {
            Bloqueada = true;
            Console.WriteLine($"Puerta número {NumPuerta} bloqueada automáticamente.");
        }

        public override void desbloquear()
        {
            Bloqueada = false;
            Console.WriteLine($"Puerta número {NumPuerta} desbloqueada automáticamente.");
        }

        public void Timeout()
        {
            if (!Bloqueada)
            {
                desbloquear();
            }
            else { bloquear(); }
        }
    }

}

File: DomotizacionInterfaces/src/main/java/org/example/PuertaAutomatica.java
package org.example;

public class PuertaAutomatica extends Puerta{

    protected PuertaAutomatica(int numPuerta) {
        super(numPuerta);
    }

    public void bloquear()
    {
        setBloqueada(true);
        System.out.println("Puerta número " + getNumPuerta() + " bloqueada automáticamente");
    }

    public void desbloquear()
    {
        setBloqueada(false);
        System.out.println("Puerta número " + getNumPuerta() + " desbloqueada automáticamente.");
    }

    public void Timeout()
    {
        if (!isBloqueada())
        {
            desbloquear();
        }
        else { bloquear(); }
    }


}
